critiquebrainz/frontend/external/bookbrainz_db/edition_group.py

```diff
@@ -0,0 +1,77 @@
+import uuid
+from brainzutils import cache
+from typing import List
+import sqlalchemy
+import critiquebrainz.frontend.external.bookbrainz_db as db 
+from critiquebrainz.frontend.external.bookbrainz_db import DEFAULT_CACHE_EXPIRATION
+from critiquebrainz.frontend.external.bookbrainz_db.identifiers import fetch_identifiers
+from critiquebrainz.frontend.external.bookbrainz_db.relationships import fetch_relationships
+
+
+def get_edition_group_by_bbid(bbid: uuid.UUID) -> dict:
+    """
+    Get info related to an edition group using its BookBrainz ID.
+    Args:
+        bbid : BBID of the edition group.
+    Returns:
+        A dictionary containing the basic information related to the edition group.
+    """
+    edition_group = fetch_multiple_edition_groups([bbid])
+    if not edition_group:
+        return None
+    return edition_group[bbid]
```

Comment:
there's a very sneaky bug here. 
If you pass in a bbid formatted slightly differently (e.g. "343ca0d1-cf7e-4a69-8701-97733d81c3B0" - note that the last b is upper-case), then postgres will happily accept it, but return `edition_group.bbid` in lower-case, and this dictionary key access will fail.

In AcousticBrainz, we check user-provided data at the API level by turning it into a UUID object and then back into a string: https://github.com/metabrainz/acousticbrainz-server/blob/ed51bcc5afaf07f78fd0cdf48fc28db844c3ca6b/webserver/views/api/v1/core.py#L196-L200
however it's much more common to do this at the user-level, and assume that the data is clean by the time it gets down to a low-level data method like this one. I see that the musicbrainz methods fail in the same way.
I'll take a deeper look into this and let you know if we should change it.